docs/user-guide/deployments.md

```diff
@@ -454,6 +454,133 @@ nginx-deployment-3066724191   0         0         1h
 Note: You cannot rollback a paused Deployment until you resume it.
 
 
+## Failed Deployment
+
+There will be times your Deployment will get stuck trying to deploy its newest ReplicaSet without ever completing,
+due to insufficient quota, readiness probe failures, image pull error, insufficient permissions, limit ranges, or any
+application runtime misconfiguration. In those cases you can specify a timeout parameter in the spec
+([spec.progressDeadlineSeconds](#progress-deadline-seconds)) that denotes the seconds you want to wait for your Deployment
+to report any progress. Once the deadline is exceeded, meaning your Deployment will not report any progress for more than
+progressDeadlineSeconds, the deployment controller will add a condition with `Type=Progressing`, `Status=False`, and
+`Reason=ProgressDeadlineExceeded` in the status of the Deployment. For more information on status conditions, see
+[here](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/devel/api-conventions.md#typical-status-properties).
+
+Note that at the moment, Kubernetes will do nothing more than just report with a condition with `Reason=ProgressDeadlineExceeded`.
```

Comment:
We do not talk about "future features" in our technical docs.  It makes tracking changes really hard. Please change this paragraph, or at least remove the sentence starting with "In the future..."